### ISSUE TITLE
How to do profiling with 

### ISSUE THREAD

**Issue by Funkwai** [Microsoft/ChakraCore#5541]
I discovered that ChakraCore has supported ETW (https://github.com/Microsoft/ChakraCore/pull/1986
, https://github.com/Microsoft/ChakraCore/issues/328)

I want to leverage this powerful tool for CPU profiling and further analysis, for example, which is the hotspot in JS program/runtime, but I still can't find a way to get useful profile data.

According to the ETW introduction here (https://channel9.msdn.com/Events/Build/2017/P4099), I open MS TraceView, create a new LogSession with a particular GUID "{57277741-3638-4A4B-BDBA-0AC6E45DA56C}" (manifests/Microsoft-Scripting-Chakra-Instrumentation.man), but I see a lot of event activities logging in TraceView even I haven't launch ch.exe, I don't know why but I guess maybe conflict with the ChakraCore in my Edge browser due to they share the same GUID, is it possible?

Therefore, I changed the last digit of GUID from "C" to "B", then TraceView seems able to get some runtime behavior data from the particular ChakraCore during the javascript program execution, but these are all unknown event and process, you can check my attachment for the detail.

Could you tell me how to enable ETW feature in ChakraCore correctly?
![wpa](https://user-images.githubusercontent.com/22340022/43291350-eb0769a2-90e5-11e8-8ca2-8723576cff19.png)

**Comment by MSLaguana** [comment]
Yes, ChakraCore shares a GUID with Chakra (and I believe with jscript9 as well, the javascript engine in internet explorer). If you change the guid, then you'll need to register your new GUID with an appropriate manifest using `wevtutil`.

Instead of doing that, I would recommend keeping the same GUID but using WPA to filter down to only events from the process that you care about. That way WPA should already understand the events.

**Comment by Funkwai** [comment]
Thanks for your advice.
I do the profiling with ETW using the original GUID, but almost all the profile information from my process is still shown as "unknown" except the name of Javascript program, do you know why?

my command in cmd is "> {PATH}\ChakraCore\Build\VcBuild\bin\x64_debug\ch.exe -bgjit- box2d.js"

![wpa2](https://user-images.githubusercontent.com/22340022/43304618-aeb381e0-9129-11e8-9fff-423070615f44.png)

besides, ch.exe can't run in background mode with ETW at the same time, will get the following error "ASSERTION 18040: (chakracore\lib\runtime\base\functionbody.cpp, line 128) !jobProcessorCS || !jobProcessor->ProcessesInBackground() || !jobProcessorCS->IsLocked()
 Failure: (!jobProcessorCS || !jobProcessor->ProcessesInBackground() || !jobProcessorCS->IsLocked())
FATAL ERROR: ch.exe failed due to exception code c0000420"

**Comment by digitalinfinity** [comment]
Hi @Funkwai - can you remind me how you're collecting the ETW trace again? It looks like you're missing loader events- if you use WPRUI, you can check Edge Browser and that should get you a useable trace

![image](https://user-images.githubusercontent.com/158390/43337281-3b15aeb6-9188-11e8-9fe2-eb5bf474b018.png)

**Comment by sethbrenith** [comment]
You might also need to do "Configure symbol paths" and add whatever folder is holding chakracore.pdb, so that the "Load symbols" button can then find it.

**Comment by Funkwai** [comment]
@MSLaguana @sethbrenith Thanks, the trace log looks great after the symbols are loaded, the internal function of ChakraCore and JITed functions are shown properly.

A further question, is this profiling method with ETW also supported in win10-arm platform? I downloaded WPR & WPA on ASUS NovaGo TP370QL (Arm-processor based) laptop and do the same profiling process, but occur an unexpectedly quit in WPR during saving a trace to storage. For the detail, you can watch the last few seconds of [this video](https://youtu.be/bqCGRAEJ9Zw). Although .etl file can be generated in the end, it didn't show any usable data on WPA.
Any idea what might be going on here?
